tests/test_passes.py

```diff
@@ -733,6 +732,68 @@ def test_nested_elimination(self):
         pyrtl.working_block().sanity_check()
 
 
+class TestSynthPasses(NetWireNumTestCases):
+    in1: pyrtl.Input
+    in2: pyrtl.Input
+    out: pyrtl.Output
+
+    def setUp(self):
+        pyrtl.reset_working_block()
+        self.in1 = pyrtl.Input(bitwidth=5, name='in1')
+        self.in2 = pyrtl.Input(bitwidth=5, name='in2')
+        self.out = pyrtl.Output(bitwidth=5, name='out')
+
+    def run_nand_synth(self, values: dict[tuple[int, int], int]):
+        pyrtl.synthesize()
+        pyrtl.nand_synth()
+        sim_trace = pyrtl.SimulationTrace()
+        sim = pyrtl.Simulation(tracer=sim_trace)
+        for inputs, output in values.items():
+            sim.step({'in1': inputs[0], 'in2': inputs[1]})
+            self.assertEqual(sim.inspect('out'), output, msg=f"Failed on inputs {inputs[0]} and {inputs[1]}")
+
+    def test_nand_synth_and(self):
+        self.out <<= self.in1 & self.in2
+        self.run_nand_synth({(1, 2): 0, (4, 5): 4, (7, 11): 3})
```

Comment:
Consider writing these `values` like
`{(1, 2): 1 & 2, (4, 5): 4 & 5, (7, 11): 7 & 11}`
so the mapping from inputs to outputs is a little clearer